flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/BasicTypeComparator.java

```diff
@@ -42,6 +42,10 @@ protected BasicTypeComparator(boolean ascending) {
 		this.ascendingComparison = ascending;
 	}
 
+	public boolean isAscendingComparison() {
```

Comment:
There is no need to touch this class, if you overwrite the `createComparator` method in the `RowTypeInfo` (details below).